### PR TITLE
Season 1

### PR DESCRIPTION
**Pull request by Pausbe**
## Changelog

**- Added Season 1 Weapons**
- **Sniper Rifle:** Victus XMR (Launch)
- **Submachine Gun:** BAS-P (Launch)
- **Assault Rifle:** M13B (Launch)
- **Assault Rifle:** Chimera (Mid-Season)

**Note:** Since this game is based around camo challenges it's hard to update the tracker in advance without knowing the actual challenges for the camos. What's the best course of action for this? We can leave camos as placeholders until we know what the camo challenges are, but if users are already beginning to track, then it will reset when we update the camo name. Or we can wait until we get all the information to push an update, but on the downside, users have to wait to actually track their camos. Thoughts?

### CODE REVIEW

**Comment by carlssonemil** [comment]
I think it's best to wait until we have all the details, or atleast the names of the camouflages. The challenge requirements we can add in continuously afterwards.

We could maybe add the weapon as "Coming soon", so that the users know that they are coming 🤔

File: src/data/defaults/progress/assaultRifles.js [hunk 1]
```diff
@@ -54,4 +54,18 @@ export default {
...
+		'TBA3': false,
+		'TBA4': false,
+	},
+
+	'Chimera': {
+		'TBA': false,
+		'TBA2': false,
+		'TBA3': false,
+		'TBA4': false,
+	},
```
Comment (carlssonemil) [inline src/data/defaults/progress/assaultRifles.js]: Leave empty for now.
```suggestion
	'M13B': {},

	'Chimera': {},
```

File: src/data/defaults/progress/sniperRifles.js [hunk 2]
```diff
@@ -26,4 +26,11 @@ export default {
 		'Aspen': false,
 		'Lichyard': false,
 	},
+
+	'Victus XMR': {
+		'TBA': false,
+		'TBA2': false,
+		'TBA3': false,
+		'TBA4': false,
+	},
```
Comment (carlssonemil) [inline src/data/defaults/progress/sniperRifles.js]:
```suggestion
	'Victus XMR': {},
```

File: src/data/defaults/progress/subMachineGuns.js [hunk 3]
```diff
@@ -54,4 +54,11 @@ export default {
 		'Feral Beast': false,
 		'2087': false,
 	},
+
+	'BAS-P': {
+		'TBA': false,
+		'TBA2': false,
+		'TBA3': false,
+		'TBA4': false,
+	},
```
Comment (carlssonemil) [inline src/data/defaults/progress/subMachineGuns.js]:
```suggestion
	'BAS-P': {},
```

File: src/data/requirements/weapons/assaultRifles.js [hunk 4]
```diff
@@ -82,4 +82,20 @@ export default {
...
+		...masteryChallenges,
+	},
+
+	'Chimera': {
+		'TBA': woodland['TBA'],
+		'TBA2': tiger['TBA'],
+		'TBA3': stripes['TBA'],
+		'TBA4': digital['TBA'],
+		...masteryChallenges,
+	},
```
Comment (carlssonemil) [inline src/data/requirements/weapons/assaultRifles.js]:
```suggestion
	'M13B': {},

	'Chimera': {},
```

File: src/data/requirements/weapons/sniperRifles.js [hunk 5]
```diff
@@ -48,4 +48,12 @@ export default {
...
 		...masteryChallenges,
 	},
+	
+	'Victus XMR': {
+		'TBA': sprayPaint['TBA'],
+		'TBA2': foliage['TBA'],
+		'TBA3': foliage['TBA'],
+		'TBA4': skulls['TBA'],
+		...masteryChallenges,
+	},
```
Comment (carlssonemil) [inline src/data/requirements/weapons/sniperRifles.js]:
```suggestion
	'Victus XMR': {},
```

File: src/data/requirements/weapons/subMachineGuns.js [hunk 6]
```diff
@@ -83,4 +83,12 @@ export default {
...
 		...masteryChallenges,
 	},
+
+	'BAS-P': {
+		'TBA': geometric['TBA'],
+		'TBA2': tiger['TBA'],
+		'TBA3': tiger['TBA'],
+		'TBA4': classic['TBA'],
+		...masteryChallenges,
+	},
```
Comment (carlssonemil) [inline src/data/requirements/weapons/subMachineGuns.js]:
```suggestion
	'BAS-P': {},
```

File: src/data/weapons/assaultRifles.js [hunk 7]
```diff
@@ -11,7 +11,7 @@ const base = [
 	'Kastov 545',
 ]
 
-const dlc = []
+const dlc = ['M13B', 'Chimera']
```
Comment (carlssonemil) [inline src/data/weapons/assaultRifles.js:14]:
```suggestion
const dlc = ['M13B', 'Chimera']

const comingSoon['M13B', 'Chimera']
```

And then add a property below on the weapon object:

```javascript
comingSoon: comingSoon.includes(weapon)
```

File: src/data/weapons/sniperRifles.js [hunk 8]
```diff
@@ -2,7 +2,7 @@ import defaultSniperRiflesProgress from '../defaults/progress/sniperRifles'
 
 const base = ['MCPR-300', 'Signal 50', 'LA-B 330', 'SP-X 80']
 
-const dlc = []
+const dlc = ['Victus XMR']
```
Comment (carlssonemil) [inline src/data/weapons/sniperRifles.js:5]:
```suggestion
const dlc = ['Victus XMR']

const comingSoon = ['Victus XMR']
```

File: src/data/weapons/subMachineGuns.js [hunk 9]
```diff
@@ -11,7 +11,7 @@ const base = [
 	'Fennec 45',
 ]
 
-const dlc = []
+const dlc = ['BAS-P']
```
Comment (carlssonemil) [inline src/data/weapons/subMachineGuns.js:14]:
```suggestion
const dlc = ['BAS-P']

const comingSoon = ['BAS-P']
```

**Review by carlssonemil — CHANGES_REQUESTED** [review]

**Comment by carlssonemil** [comment]
I'll add a "Coming soon" label to the UI before this is merged 😃

**Comment by Pausbe** [comment]
Added the camos and camo challenges for the Season 1 Launch Weapons!

**Review by carlssonemil — APPROVED** [review]
Looks good! You got a few merge conflicts it seems like 🤔 Try rebasing the main branch into your branch locally and try and fix the conflicts there!

**Comment by Pausbe** [comment]
I think I did it, I hope I didn't break anything 😅

**Review by carlssonemil — APPROVED** [review]
Looks good to me! 🥳

**Pull request merged**